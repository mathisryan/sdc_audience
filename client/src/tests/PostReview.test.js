import React from 'react';
import { shallow, mount } from 'enzyme';
import PostReview from '../components/PostReview';
import './setupTest'

describe('PostReview', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<PostReview debug />);
    expect(component).toMatchSnapshot();
  });
  
  it('should have correct rating text', () => {
    const wrapper = mount(<PostReview/>);
    expect(wrapper.find('span').at(0).text()).toEqual('Add your rating');
  });
});
