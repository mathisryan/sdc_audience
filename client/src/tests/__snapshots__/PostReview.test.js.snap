// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostReview should render correctly in "debug" mode 1`] = `
<div
  className="adduser-container"
>
  <div
    className="post-container"
  >
    <div
      className="addRating"
    >
      <span
        className="ratingText"
      >
        Add your rating
      </span>
    </div>
    <div
      className="rating"
    >
      <div
        className="profile box"
      >
        <i
          className="far fa-user fa-5x"
        />
      </div>
      <div
        className="ratingReview"
      >
        <div
          className="add-wrapper"
        >
          <button
            className="interested box"
            value="-"
          >
            - Not interested
          </button>
        </div>
        <div
          className="add-wrapper"
        >
          <button
            className="wantToSee box"
            value="+"
          >
            + Want to See
          </button>
        </div>
        <div
          className="starsRating box"
        >
          <span
            className="fa fa-star"
            data-rating="1"
          />
          <span
            className="fa fa-star"
            data-rating="2"
          />
          <span
            className="fa fa-star"
            data-rating="3"
          />
          <span
            className="fa fa-star"
            data-rating="4"
          />
          <span
            className="fa fa-star"
            data-rating="5"
          />
          <input
            className="rating-value"
            name="whatever"
            type="hidden"
            value="3"
          />
        </div>
        <div
          className="review-wrapper"
        >
          <textarea
            className="addReview box"
            onChange={[Function]}
            value=""
          >
            Add a Review
          </textarea>
          <div
            className="rating-submit"
          >
            <button
              className="post btn"
            >
              Post
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
